src/rust/src/x509/sct.rs

```diff
@@ -175,17 +174,19 @@ impl Sct {
 
     #[getter]
     fn timestamp<'p>(&self, py: pyo3::Python<'p>) -> pyo3::PyResult<&'p pyo3::PyAny> {
+        let utc = types::DATETIME_TIMEZONE_UTC.get(py)?;
+
+        let kwargs = pyo3::types::PyDict::new(py);
+        kwargs.set_item("microsecond", self.timestamp % 1000 * 1000)?;
+        kwargs.set_item("tzinfo", None::<Option<pyo3::PyAny>>)?;
+
         types::DATETIME_DATETIME
             .get(py)?
             .call_method1(
-                pyo3::intern!(py, "utcfromtimestamp"),
-                (self.timestamp / 1000,),
+                pyo3::intern!(py, "fromtimestamp"),
+                (self.timestamp / 1000, utc),
             )?
-            .call_method(
-                "replace",
-                (),
-                Some(vec![("microsecond", self.timestamp % 1000 * 1000)].into_py_dict(py)),
-            )
+            .call_method("replace", (), Some(kwargs))
```

Comment:
NB: We have to do this song-and-dance because `fromtimestamp` creates an aware `datetime` when given a timezone, which breaks the current API promise (a naive `datetime`).

If we don't pass it a timezone it "cleverly" adjusts the given timestamp to the local timezone and returns it as a naive `datetime`, which is also incorrect. Instead, we create it as an aware `datetime` and update the pre-existing call to `replace` to make it naive after the fact.

This is the aware-to-naive conversion mechanism documented (encouraged?) by Python itself: https://docs.python.org/3/library/datetime.html#datetime.datetime.replace